api/repositories/route_repository.go

```diff
@@ -381,9 +398,17 @@ func (f *RouteRepo) AddDestinationsToRoute(ctx context.Context, authInfo authori
 	cfRoute := baseCFRoute.DeepCopy()
 	cfRoute.Spec.Destinations = mergeDestinations(message.ExistingDestinations, message.NewDestinations)
 
-	err := f.privilegedClient.Patch(ctx, cfRoute, client.MergeFrom(baseCFRoute))
-	if err != nil { // untested
-		return RouteRecord{}, fmt.Errorf("err in client.Patch: %w", err)
+	userClient, err := f.userClientFactory.BuildClient(authInfo)
+	if err != nil {
+		return RouteRecord{}, fmt.Errorf("failed to build user client: %w", err)
+	}
+
+	err = userClient.Patch(ctx, cfRoute, client.MergeFrom(baseCFRoute))
+	if err != nil {
+		if apierrors.IsForbidden(err) {
+			return RouteRecord{}, NewForbiddenError(RouteResourceType, err)
+		}
+		return RouteRecord{}, fmt.Errorf("err in client.Patch: %w", err) // untested
```

Comment:
It looks like none of the repository changes are being tested in the `route_repository_test`. While we're covered by the higher level integration tests, I'd still like to see that test covering the whole functionality of the repository.